From 2787a75163061f95bfd8e25516fca7c83624baa2 Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@gmail.com>
Date: Tue, 5 May 2015 15:46:18 -0700
Subject: [PATCH 0485/1164] CDH-36943: HIVE-10213 : MapReduce jobs using
 dynamic-partitioning fail on commit (Mithun
 Radhakrishnan via Sushanth Sowmyan)

Change-Id: Ia30bfcc3dfaddd74d5b58da04804ba236f4f3fc5
---
 .../mapreduce/TaskCommitContextRegistry.java       |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/TaskCommitContextRegistry.java b/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/TaskCommitContextRegistry.java
index 8c6d0fc..bdffb19 100644
--- a/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/TaskCommitContextRegistry.java
+++ b/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/TaskCommitContextRegistry.java
@@ -71,7 +71,8 @@ public static TaskCommitContextRegistry getInstance() {
   public synchronized void commitTask(TaskAttemptContext context) throws IOException {
     String key = generateKey(context);
     if (!taskCommitters.containsKey(key)) {
-      throw new IOException("No callback registered for TaskAttemptID:" + key);
+      LOG.warn("No callback registered for TaskAttemptID:" + key + ". Skipping.");
+      return;
     }
 
     try {
@@ -99,7 +100,8 @@ private String generateKey(TaskAttemptContext context) throws IOException {
   public synchronized void abortTask(TaskAttemptContext context) throws IOException {
     String key = generateKey(context);
     if (!taskCommitters.containsKey(key)) {
-      throw new IOException("No callback registered for TaskAttemptID:" + key);
+      LOG.warn("No callback registered for TaskAttemptID:" + key + ". Skipping.");
+      return;
     }
 
     try {
-- 
1.7.9.5

